app/services/foreman/plugin.rb

```diff
@@ -245,12 +246,26 @@ def pending_migrations
 
     # List of helper methods allowed for templates in safe mode
     def allowed_template_helpers(*helpers)
-      Foreman::Renderer::ALLOWED_HELPERS.concat(helpers)
+      in_to_prepare do
+        Foreman::Renderer::ALLOWED_HELPERS.concat(helpers).uniq!
+      end
     end
 
     # List of variables allowed for templates in safe mode
     def allowed_template_variables(*variables)
-      Foreman::Renderer::ALLOWED_VARIABLES.concat(variables)
+      in_to_prepare do
+        Foreman::Renderer::ALLOWED_VARIABLES.concat(variables).uniq!
+      end
+    end
+
+    # List of modules which public methods will be available during template rendering
+    # including safe mode
+    def extend_template_helpers(*mods)
+      mods.each do |mod|
+        in_to_prepare do
```

Comment:
Some micro-optimization: swapping the loops would lead to a bit less blocks saved in the callbacks